src/server/game/Entities/Pet/Pet.cpp

```diff
@@ -135,18 +144,67 @@ bool Pet::LoadPetFromDB(Player* owner, uint32 petEntry, uint32 petnumber, bool c
         stmt->setUInt8(2, uint8(PET_SAVE_LAST_STABLE_SLOT));
     }
 
-    result = CharacterDatabase.Query(stmt);
+    WorldSession* session = owner->GetSession();
+    Pet* pet = this;
+    session->GetQueryProcessor().AddQuery(CharacterDatabase.AsyncQuery(stmt)
+        .WithChainingPreparedCallback([pet, session, callback, current](QueryCallback& queryCallback, PreparedQueryResult result)
+    {
+        Player* player = session->GetPlayer();
+        if (!player || !player->IsInWorld())
+        {
+            delete pet;
```

Comment:
it looks quite dangerous to me to delete this, changing the usual common handling of LoadFromDB() result.

Reply:
Just wondering who won in this argument:
@jackpoz vs @ariel-  or @tripleslash suggestion